src/core/service.c

```diff
@@ -1439,6 +1439,48 @@ static bool service_exec_needs_notify_socket(Service *s, ExecFlags flags) {
         return s->notify_access != NOTIFY_NONE;
 }
 
+static Service *service_get_triggering_service(Service *s) {
+        assert(s);
+
+        /* Return the service which triggered service 's', this means dependency
+         * types which include the UNIT_ATOM_PROPAGATE_EXIT_STATUS atom.
+         *
+         * N.B. if there are multiple services which could trigger 's' via OnFailure=
+         * or OnSuccess= then we return NULL. This is since we don't know from which
+         * one to propagate the exit status.
+         */
+
+        Unit *candidate = NULL;
+        Unit *other;
+
+        UNIT_FOREACH_DEPENDENCY(other, UNIT(s), UNIT_ATOM_PROPAGATE_EXIT_STATUS) {
+                assert(other->type == UNIT_SERVICE);
+
+                if (unit_has_dependency(other, UNIT_ATOM_ON_SUCCESS, UNIT(s)) ||
+                    unit_has_dependency(other, UNIT_ATOM_ON_FAILURE, UNIT(s))) {
+                    if (candidate) {
+                        log_unit_debug(UNIT(s), "multiple trigger source candidates for exit status propagation");
+                        return NULL;
+                    } else
+                        candidate = other;
```

Comment:
something went wrong with the indentation here...

Reply:
Yes,not sure what happened. Will fix.